### PR TITLE
Update Travis to test latest golang versions.

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 language: go
-go: 1.2
+go:
+    - 1.6.3
+before_install:
+    - gotools=golang.org/x/tools
 install:
-    - go get -v code.google.com/p/go.tools/cmd/cover
+    - go get -v $gotools/cmd/cover
```
Comment (shurcooL) [inline .travis.yml]: Why is the `gotools` variable needed, instead of just doing `go get -v golang.org/x/tools/cmd/cover` here?

**Comment by davecgh** [comment]
There is infastructure to support more than one possible desired test outcome:

The following diff should allow the tests to work on the latest Go with the renamed type as well.

```diff
--- a/spew/dumpcgo_test.go
+++ b/spew/dumpcgo_test.go
@@ -59,10 +59,11 @@ func addCgoDumpTests() {
        v3Len := fmt.Sprintf("%d", v3l)
        v3Cap := fmt.Sprintf("%d", v3c)
        v3t := "[6]testdata._Ctype_unsignedchar"
+       v3t2 := "[6]testdata._Ctype_uchar"
        v3s := "(len=" + v3Len + " cap=" + v3Cap + ") " +
                "{\n 00000000  74 65 73 74 33 00                               " +
                "  |test3.|\n}"
-       addDumpTest(v3, "("+v3t+") "+v3s+"\n")
+       addDumpTest(v3, "("+v3t+") "+v3s+"\n", "("+v3t2+") "+v3s+"\n")

        // C signed char array.
        v4, v4l, v4c := testdata.GetCgoSignedCharArray()
```

File: .travis.yml (hunk 1 above)
Comment (davecgh) [inline .travis.yml]: Agree with @shurcooL here.  Please just use `go get -v golang.org/x/tools/cmd/cover` here and remove the `before_install` bits.

**Comment by davecgh** [comment]
OK as soon as the tests pass.  Thanks!

**Pull request merged**